streams/src/main/java/org/apache/kafka/streams/kstream/Window.java

```diff
@@ -56,9 +56,7 @@ public long end() {
      * @param other  another window
      * @return       {@code true} if {@code other} overlaps with this window&mdash;{@code false} otherwise
      */
-    public boolean overlap(Window other) {
-        return this.start() < other.end() || other.start() < this.end();
-    }
+    public abstract boolean overlap(Window other);
```

Comment:
Does this need to be abstract? Can't we just have the impl of it here? i.e., the `SessionWindow` and `TimeWindow` implementations are identical.